example/components/PubNubTime.tsx

```diff
@@ -1,17 +1,22 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { usePubNub } from '../../src/index';
 
 const PubNubTime = () => {
-  const PubNubClient = usePubNub();
+  const client = usePubNub();
+  const [time, setTime] = useState<number | null>(null);
 
-  return (
-    <div>
-      {PubNubClient.time((status, response): void => {
-        if (status.error) console.log('error', status);
-        else console.log('response', response.timetoken);
-      })}
-    </div>
-  );
+  useEffect(() => {
+    client
+      .time()
+      .then(({ timetoken }) => {
+        setTime(timetoken);
+      })
+      .catch(error => {
+        console.log(error);
```

Comment:
I really like the cleanup of this example.

What do you think about showing the error to the component output if it occurs.  I don't feel strongly about it one way or another, but it might be helpful is someone actually tries it and does get an error.

Along the same lines, is there a useful thing we might show prior to fetching the time so that someone actually trying it would see when the time arrived?

Reply:
We can show a loading state based on null timetoken. I will include this!